src/modules/RA/DataLayer/components/tabs/StyleTab/Style/CategorizeValue.js

```diff
@@ -0,0 +1,99 @@
+
+import React from 'react';
+
+import { useTranslate } from 'react-admin';
+import { makeStyles } from '@material-ui/core/styles';
+import FormLabel from '@material-ui/core/FormLabel';
+import { Field, useField } from 'react-final-form';
+
+import { usePrevious } from '../../../../../../../utils/hooks';
+
+import styles from './styles';
+
+const useStyles = makeStyles(styles);
+
+const DEFAULT_MAX_CATEGORIES = 20;
+
+const CategorizeValue = ({
+  path,
+  maxCategories = DEFAULT_MAX_CATEGORIES,
+  getValuesOfProperty,
+  Component,
+  defaultValueGenerator,
+}) => {
+  const translate = useTranslate();
+  const classes = useStyles();
+  const [tooManyValues, setTooManyValues] = React.useState(false);
+  const { input: { value: field } } = useField(`${path}.field`);
+  const { input: { value: valueList, onChange: setValueList } } = useField(`${path}.categories`);
+
+  const prevField = usePrevious(field);
+
+  React.useEffect(() => {
+    const getValueList = async () => {
+      if (prevField === field) return;
+      setTooManyValues(false);
+      const result = await getValuesOfProperty(field);
+
+      if (result.length > maxCategories) {
+        setTooManyValues(true);
+        return;
```

Comment:
Maybe you have to check if this component is still mounted before executing a setState